applications/CoSimulationApplication/python_scripts/solver_wrappers/external/flux_wrapper.py

```diff
@@ -0,0 +1,712 @@
+
+        This function advances the solution in time based on the scenario type defined in the JSON settings.
+
+        Args:
+            self: An instance of the FluxWrapper class.
+            current_time: The current time.
+
+        Returns:
+            float: The new time after the time advancement.
+
+        Note:
+            The behavior of this method depends on the scenario type specified in the settings.
+        """
+
+        # In  the case of current_time = -1.0
+        if current_time < 0.0 :
+            return 0.0
+
+        # Get the Flux scenario type from JSON file
+        scenario_type = self.settings["solver_wrapper_settings"]["scenario_type"].GetString()
+        if scenario_type == "single_step" :
+            # Nothing to do in the case
+            return 0.0
+        elif scenario_type == "time_avg" :
+            # Delete all solved steps to reactivate first step of the scenario
+            self.flux.DeleteAllSolvedSteps()
+            # No pertinent active time
+            return 0.0
+        elif scenario_type == "time_inst" :
+            self.flux.ActivateNextStep()
+            self.flux.InitializeStep()
+            new_time = self.flux.GetCurrentTime()
+            if current_time == None : return 0.0
+            else : return new_time
+
+    def SolveSolutionStep(self):
+        """
+        Solves the current time step including export/import processes.
+
+        Args:
+            self: An instance of the class.
+        """
+
+        # ------------------------
+        # Solve the required steps
+        # ------------------------
+        # Get scenario type from JSON file
+        scenario_type = self.settings["solver_wrapper_settings"]["scenario_type"].GetString()
+
+        # In the case of a single step or an instantaneous scenario
+        if scenario_type == "single_step" or scenario_type == "time_inst" :
+            # Export all data from Kratos to Flux
+            self.ExportAllData()
+            # Solve the current step
+            self.flux.SolveStep()
+            # Import all data from Flux to Kratos
+            self.ImportAllData()
+
+        # In the case of a time averaged scenario
+        elif scenario_type  == "time_avg" :
+            # If scenario is already solved
+            if not self.flux.HasNextStep() :
+                self.flux.DeleteAllSolvedSteps()
+            # Export all data from Kratos to Flux
+            self.ExportAllData()
+            # Solve all steps
+            self.flux.SolveAllSteps()
+            # Import all data from Flux to Kratos
+            self.ImportAllData()
+
+    def _GetIOType(self):
+        """
+        Get the Flux I/O type dedicated to handling import/export data.
+
+        Returns:
+            str: The string "flux_io".
+
+        Note:
+            This method specifies the dedicated I/O type for handling data import and export using Flux.
+
+        """
+
+        return "flux_io"
```

Comment:
```suggestion
```

this does nothing, let the baseclass handle it

Reply:
Okay